### PR TITLE
Task/morecast v2 api refactor

### PR DESCRIPTION
**Pull request by conbrad**
- Creates new `morecast-v2/determinates/<start-date>/<end-date>` endpoint that returns the lists of actuals and predictions, as well as all forecasts for the date range 
    - Since forecasts include update date and no station name we just return all of them
# Test Links:
[Landing Page](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2786.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Comment by dgboss** [comment]
>  the predictions returned will be all predictions that have dates strictly after the last actual.

I think we actually need all predictions for the entire date range, not just those for dates greater than the last date for which there is an actual (based on this UI)
image

File: api/app/routers/morecast_v2.py [hunk 1]
```diff
@@ -146,3 +150,41 @@ async def get_observed_dailies(start_date: date, end_date: date, request: Observ
...
+
+    unique_station_codes = list(set(request.stations))
+
+    vancouver_tz = pytz.timezone("America/Vancouver")
+
+    start_time = vancouver_tz.localize(datetime.combine(start_date, time.min))
+    end_time = vancouver_tz.localize(datetime.combine(end_date, time.max))
+
+    with get_read_session_scope() as db_session:
+        forecasts: List[MoreCastForecastOutput] = get_forecasts(db_session, start_time, end_time, request.stations)
```
Comment (dgboss) [inline api/app/routers/morecast_v2.py:173]: Do you think it would be possible to reshape forecast objects to match `WeatherIndeterminate`?

Comment (dgboss) [reply api/app/routers/morecast_v2.py:173]: Oh wait, I guess we're not consuming these on the front end yet, so no need at the moment.

File: api/app/schemas/morecast_v2.py [hunk 2]
```diff
@@ -91,3 +103,22 @@ class ObservedDaily(BaseModel):
...
     """ Yesterday station dailies response """
     dailies: List[ObservedDaily]
+
+
+class WeatherIndeterminate(BaseModel):
+    """ Used to represent a predicted or actual value """
+    station_code: int
+    station_name: str
+    determinate: WeatherDeterminate
+    utcTimestamp: datetime
```
Comment (dgboss) [inline api/app/schemas/morecast_v2.py:113]: Should `utcTimestamp`be snake case?

**Review by dgboss — APPROVED** [review]

File: api/app/routers/morecast_v2.py (hunk 1 above)
Comment (conbrad) [reply api/app/routers/morecast_v2.py:173]: Yeah I was gonna hold off on doing that until we start using them. Also there's a couple differences a forecast has currently: no station name and it has an update date timestamp.

File: api/app/schemas/morecast_v2.py (hunk 2 above)
Comment (conbrad) [reply api/app/schemas/morecast_v2.py:113]: Yup it should, not sure why the linter doesn't catch that

Comment (conbrad) [reply api/app/schemas/morecast_v2.py]: Fixed in: https://github.com/bcgov/wps/pull/2786/commits/128d1b9198f50d6f81a984bf58f3cb3c5a195ad8

File: api/app/routers/morecast_v2.py [hunk 3]
```diff
@@ -146,3 +150,41 @@ async def get_observed_dailies(start_date: date, end_date: date, request: Observ
...
+
+@router.post('/determinates/{start_date}/{end_date}',
+             response_model=IndeterminateDailiesResponse)
+async def get_determinates_for_date_range(start_date: date,
+                                          end_date: date,
+                                          request: StationsRequest):
+    """ Returns the weather values for any actuals, predictions and forecasts for the 
+    requested stations within the requested date range.
+    """
+    logger.info('/morecast-2/determinates/%s/%s', start_date, end_date)
```
Comment (andrea-williams) [inline api/app/routers/morecast_v2.py:163]:
```suggestion
    logger.info('/morecast-v2/determinates/%s/%s', start_date, end_date)
```

**Review by andrea-williams — APPROVED** [review]
🔥

**Pull request merged**